hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/components/AbstractNotificationView.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2011-2016 Bosch Software Innovations GmbH, Germany. All rights reserved.
+ */
+package org.eclipse.hawkbit.ui.components;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+import org.eclipse.hawkbit.repository.event.TenantAwareEvent;
+import org.eclipse.hawkbit.ui.common.table.BaseEntityEventType;
+import org.eclipse.hawkbit.ui.common.table.BaseUIEntityEvent;
+import org.eclipse.hawkbit.ui.menu.DashboardMenuItem;
+import org.eclipse.hawkbit.ui.push.EventContainer;
+import org.vaadin.spring.events.EventBus;
+import org.vaadin.spring.events.EventScope;
+import org.vaadin.spring.events.annotation.EventBusListenerMethod;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.vaadin.navigator.View;
+import com.vaadin.navigator.ViewChangeListener.ViewChangeEvent;
+import com.vaadin.ui.VerticalLayout;
+
+/**
+ * Abstract view for all views, which show notifications.
+ */
+public abstract class AbstractNotificationView extends VerticalLayout implements View {
+
+    private static final long serialVersionUID = 1L;
+    private final transient Cache<BaseUIEntityEvent<?>, Object> skipUiEvents = CacheBuilder.newBuilder()
+            .expireAfterAccess(10, SECONDS).build();
+
+    private final transient EventBus.UIEventBus eventBus;
+
+    private transient Map<Class<?>, RefreshableContainer> supportedEvents;
+
+    private final NotificationUnreadButton notificationUnreadButton;
+
+    private int viewUnreadNotifcations;
+
+    /**
+     * Constructor.
+     * 
+     * @param eventBus
+     *            the ui event bus
+     * @param notificationUnreadButton
+     *            the notificationUnreadButton
+     */
+    public AbstractNotificationView(final EventBus.UIEventBus eventBus,
+            final NotificationUnreadButton notificationUnreadButton) {
+        this.eventBus = eventBus;
+        this.notificationUnreadButton = notificationUnreadButton;
+    }
+
+    @EventBusListenerMethod(scope = EventScope.UI)
+    void onEventContainerEvent(final EventContainer<?> eventContainer) {
+        if (!supportNotificationEventContainer(eventContainer.getClass()) || eventContainer.getEvents().isEmpty()) {
+            return;
+        }
+
+        eventContainer.getEvents().stream().filter(event -> !anyEventMatch(event)).forEach(event -> {
+            notificationUnreadButton.incrementUnreadNotification(this, eventContainer);
+            viewUnreadNotifcations++;
```

Comment:
The use of `viewUnreadNotifcations` is not thread safe => `onEventContainerEvent()` will be called from different threads